src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -1580,24 +1578,87 @@ bool Graph::InsertNode(NodePtr parent, NodePtr child, NodePtr node, int parentPo
     return true;
 }
 
+#ifdef CPU_DEBUG_CAPS
+/*
+ * important debugging tools for accuracy issues
+ * define F16SET as comma separated list of node types for which F16 is allowed to be enabled
+ * define F16CNT as number of nodes totally allowed to enable f16
+ * increasing/decreasing F16CNT until accuracy issue happens/disappears
+ * from the log we can spot the first node having issue when enabled f16
+ */
+struct FP16Debug {
+    std::string safe_getenv(const char* name, const char* default_value = "") {
+        std::string value = default_value;
+        const char* p = std::getenv(name);
+        if (p)
+            value = p;
+        return value;
+    }
+
+    std::string nodeTypes = safe_getenv("F16SET", "");
+    int count_limit = atoi(safe_getenv("F16CNT", "9999999").c_str());
+    int count = 0;
+
+    bool enabled(std::string type, std::string name) {
+        if (nodeTypes.find(type + ",") != std::string::npos) {
+            if (count < count_limit) {
+                std::cout << " enabled [" << count << "/" << count_limit << "] : " << type << " " << name << std::endl;
+                count++;
+                return true;
+            }
+        }
+        return false;
+    }
+};
+#endif
+
 // Set all non const data paths precision to BF16
-void Graph::EnforceBF16() {
+void Graph::EnforceInferencePrecision() {
+    CPU_DEBUG_CAP_ENABLE(static FP16Debug f16debug);
+    auto inferPrec = InferenceEngine::Precision::FP32;
+    switch (getConfig().inferencePrecision) {
+    case ov::element::bf16:
+        inferPrec = InferenceEngine::Precision::BF16;
+        break;
+    case ov::element::f16:
+        inferPrec = InferenceEngine::Precision::FP16;
+        break;
+    default:
+        break;
+    }
+
+    if (inferPrec == InferenceEngine::Precision::FP32)
```

Comment:
Maybe just do return in `default:` case in switch block above?

Reply:
OK, will change